src/kOS.Safe/Communication/BaseMessage.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using kOS.Safe.Encapsulation;
+using kOS.Safe.Serialization;
+using kOS.Safe.Exceptions;
+using kOS.Safe;
+
+namespace kOS.Safe.Communication
+{
+    public class BaseMessage : IDumper, IComparable<BaseMessage>
+    {
+        public const string DumpSentAt = "sentAt";
+        public const string DumpReceivedAt = "receivedAt";
+        public const string DumpContent = "content";
+
+        public double SentAt { get; set; }
+        public double ReceivedAt { get; set; }
+
+        /// <summary>
+        /// Message content can be either a simple encapsulated type (something that implements ISerializableValue) or a Dump.
+        ///
+        /// Currently kOS serialization doesn't allow primitives as top level objects, but Messages do allow them.
+        /// </summary>
+        private object content;
+
+        public object Content {
+            get {
+                return content;
+            }
+            set {
+                if (value is PrimitiveStructure || value is Dump)
+                {
+                    content = value;
+                } else
+                {
+                    throw new KOSException("Message can only contain primitives and serializable types");
+                }
+            }
+        }
```

Comment:
Minor nitpick : I think everyone else on the project is using the format that puts the opening ``{`` on its own separate line.  I don't care for it myself because it introduces a lot of extra vertical whitespace and thus reduces what can fit on the screen, but that does seem to be the standard on the project.